algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/preprocessor.py

```diff
@@ -0,0 +1,557 @@
+    -6.7949142456054688,
+    -6.8634209632873535,
+    -6.94186544418335
+]
+LIBRISPEECH_STD_VECTOR = [
+    3.4353282451629639,
+    3.5962932109832764,
+    3.7012472152709961,
+    3.7369205951690674,
+    3.7535104751586914,
+    3.693629264831543,
+    3.6922497749328613,
+    3.7641522884368896,
+    3.8419716358184814,
+    3.8999848365783691,
+    3.9294240474700928,
+    3.9317409992218018,
+    3.9139585494995117,
+    3.9031598567962646,
+    3.8691999912261963,
+    3.8155081272125244,
+    3.7644970417022705,
+    3.7099106311798096,
+    3.6965086460113525,
+    3.6003766059875488,
+    3.5493226051330566,
+    3.5465121269226074,
+    3.45003604888916,
+    3.4712812900543213,
+    3.4084610939025879,
+    3.4408135414123535,
+    3.4104881286621094,
+    3.4217638969421387,
+    3.4312851428985596,
+    3.4199209213256836,
+    3.4305806159973145,
+    3.4382665157318115,
+    3.4580366611480713,
+    3.4817991256713867,
+    3.4958710670471191,
+    3.5036792755126953,
+    3.5047574043273926,
+    3.4988734722137451,
+    3.493056058883667,
+    3.4822943210601807,
+    3.459430456161499,
+    3.4612770080566406,
+    3.4559063911437988,
+    3.4755423069000244,
+    3.4971549510955811,
+    3.5326557159423828,
+    3.5705199241638184,
+    3.5920312404632568,
+    3.596907377243042,
+    3.5913500785827637,
+    3.5865931510925293,
+    3.5826809406280518,
+    3.5837743282318115,
+    3.5895791053771973,
+    3.5819313526153564,
+    3.5837869644165039,
+    3.5861184597015381,
+    3.5889589786529541,
+    3.592214822769165,
+    3.5939455032348633,
+    3.5856630802154541,
+    3.5884113311767578,
+    3.5921022891998291,
+    3.5870490074157715,
+    3.5806570053100586,
+    3.5731067657470703,
+    3.5617532730102539,
+    3.54980731010437,
+    3.5527374744415283,
+    3.5475366115570068,
+    3.5387849807739258,
+    3.5256178379058838,
+    3.5031836032867432,
+    3.4922726154327393,
+    3.4879646301269531,
+    3.4725594520568848,
+    3.4558389186859131,
+    3.4351828098297119,
+    3.4284293651580811,
+    3.4299170970916748
+]
+
+PreprocessorConfig = namedtuple(
+    "PreprocessorConfig",
+    field_names=[
+        "sample_rate",
+        "frame_size_ms",
+        "frame_step_ms",
+        "compute_energy",
+        "window_fn",
+        "output_log_floor",
+        "pad_end",
+        "preemph",
+        "preemph_htk_flavor",
+        "noise_scale",
+        "num_bins",
+        "lower_edge_hertz",
+        "upper_edge_hertz",
+        "fft_overdrive",
+        "output_floor"
+    ])
+
+
+def _hertz_to_mel(frequencies_hertz):
+  """Convert hertz to mel."""
+  if type(frequencies_hertz) in [type(0.0), type(0)]:
+    return _MEL_HIGH_FREQUENCY_Q * math.log(1.0 + (frequencies_hertz /
+                                                   _MEL_BREAK_FREQUENCY_HERTZ))
+  return _MEL_HIGH_FREQUENCY_Q * torch.log(1.0 + (frequencies_hertz /
+                                                  _MEL_BREAK_FREQUENCY_HERTZ))
```

Comment:
nit: this could probably be a bit cleaner like:

```
log_fn = math.log if type(frequencies_hertz) in [type(0.0), type(0)] else torch.log
return _MEL_HIGH_FREQUENCY_Q * log_fn(1.0 + (frequencies_hertz /
                                                  _MEL_BREAK_FREQUENCY_HERTZ))
```

Reply:
Done.